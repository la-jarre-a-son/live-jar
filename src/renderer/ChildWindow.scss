$ljas-ui-asset-path: 'assets';
$font-path: '../assets/fonts';

@import '@la-jarre-a-son/ui/lib/theme/jar/tokens';
@import '@la-jarre-a-son/ui/lib/style/normalize';
@import '@la-jarre-a-son/ui/lib/theme/jar/global';
@import '@la-jarre-a-son/ui/lib/theme/jar/fonts';
@import '@la-jarre-a-son/ui/lib/theme/jar/dark';

:global {
  body {
    position: relative;
    color: white;
    background: black;
    font-family: sans-serif;
    width: 100vw;
    height: 100vh;
    padding: 0;
    margin: 0;
    user-select: none;
  }

  #root {
    width: 100%;
    height: 100%;
  }

  iframe {
    margin: 0;
    padding: 0;
    vertical-align: top;
  }
}
