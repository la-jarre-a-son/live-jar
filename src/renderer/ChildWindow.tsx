import { ThemeProvider } from '@la-jarre-a-son/ui';

import WindowStateProvider from './contexts/WindowState';
import SettingsProvider from './contexts/Settings';

import './ChildWindow.scss';
import Stream from './views/Stream';
import StreamStatsProvider from './contexts/StreamStats';

export default function ChildWindow() {
  return (
    <ThemeProvider theme="jar" variant="dark">
      <WindowStateProvider>
        <SettingsProvider>
          <StreamStatsProvider>
            <Stream />
          </StreamStatsProvider>
        </SettingsProvider>
      </WindowStateProvider>
    </ThemeProvider>
  );
}
