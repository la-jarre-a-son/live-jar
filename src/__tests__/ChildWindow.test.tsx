import '@testing-library/jest-dom';
import { render, act } from '@testing-library/react';
import ChildWindow from '../renderer/ChildWindow';

describe('ChildWindow', () => {
  it('should render', async () => {
    const el = document.createElement('div');
    await act(async () => {
      await render(<ChildWindow />, { container: el });
    });

    expect(el).toMatchSnapshot();
  });
});
