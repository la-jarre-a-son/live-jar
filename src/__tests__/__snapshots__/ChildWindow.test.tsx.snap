// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChildWindow should render 1`] = `
<div>
  <div
    class="base"
  />
</div>
`;
